src/hotspot/share/ci/ciEnv.cpp

```diff
@@ -1185,6 +1185,26 @@ ciInstance* ciEnv::unloaded_ciinstance() {
 // Don't change thread state and acquire any locks.
 // Safe to call from VM error reporter.
 
+
+// Look up the location descriptor for the given class and return it as a string.
+// Returns the class name as a fallback if no location is found.
+const char *ciEnv::replay_name(ciKlass* k) const {
+  if (k->is_instance_klass()) {
+    return replay_name(k->as_instance_klass()->get_instanceKlass());
+  }
+  return k->name()->as_quoted_ascii();
+}
+
+// Look up the location descriptor for the given class and return it as a string.
+// Returns the class name as a fallback if no location is found.
+const char *ciEnv::replay_name(const InstanceKlass* ik) const {
+  const char* name = nullptr; // dyno_name(ik);
```

Comment:
I'd appreciate to have a comment why `dyno_name(ik)` is commented out. Maybe "JDK-8271911 is not in JDK 17, so we fall back to using the class name below."